typescript/cli/src/deploy/core.ts

```diff
@@ -115,29 +153,72 @@ export async function runCoreDeploy(params: DeployParams) {
 
 export async function runCoreApply(params: ApplyParams) {
   const { context, chain, deployedCoreAddresses, config } = params;
-  const { multiProvider } = context;
-  const evmCoreModule = new EvmCoreModule(multiProvider, {
-    chain,
-    config,
-    addresses: deployedCoreAddresses,
-  });
-
-  const transactions = await evmCoreModule.update(config);
-
-  if (transactions.length) {
-    logGray('Updating deployed core contracts');
-    for (const transaction of transactions) {
-      await multiProvider.sendTransaction(
-        // Using the provided chain id because there might be remote chain transactions included in the batch
-        transaction.chainId ?? chain,
-        transaction,
-      );
+  const { multiProvider, multiProtocolSigner } = context;
+
+  switch (multiProvider.getProtocol(chain)) {
+    case ProtocolType.Ethereum: {
+      const evmCoreModule = new EvmCoreModule(multiProvider, {
+        chain,
+        config,
+        addresses: deployedCoreAddresses,
+      });
+
+      const transactions = await evmCoreModule.update(config);
+
+      if (transactions.length) {
+        logGray('Updating deployed core contracts');
+        for (const transaction of transactions) {
+          await multiProvider.sendTransaction(
+            // Using the provided chain id because there might be remote chain transactions included in the batch
+            transaction.chainId ?? chain,
+            transaction,
+          );
+        }
+
+        logGreen(`Core config updated on ${chain}.`);
+      } else {
+        logGreen(
+          `Core config on ${chain} is the same as target. No updates needed.`,
+        );
+      }
+      break;
     }
+    case ProtocolType.CosmosNative: {
+      const signer = multiProtocolSigner?.getCosmosNativeSigner(chain) ?? null;
+      assert(signer, 'Cosmos Native signer failed!');
+
+      const cosmosNativeCoreModule = new CosmosNativeCoreModule(
+        multiProvider,
+        signer,
+        {
+          chain,
+          config,
+          addresses: deployedCoreAddresses,
+        },
+      );
 
-    logGreen(`Core config updated on ${chain}.`);
-  } else {
-    logGreen(
-      `Core config on ${chain} is the same as target. No updates needed.`,
-    );
+      const transactions = await cosmosNativeCoreModule.update(config);
+
+      if (transactions.length) {
+        logGray('Updating deployed core contracts');
+        const response = await signer.signAndBroadcast(
+          signer.account.address,
+          transactions,
+          2,
```

Comment:
What does 2 mean in the context of the fee?

Reply:
it means that it will multiply the estimated gas from the simulation by 2 to be sure it is enough to pass